sdk/include/opentelemetry/sdk/trace/tracer_provider.h

```diff
@@ -27,11 +27,14 @@ class TracerProvider final : public opentelemetry::trace::TracerProvider
    * @param sampler The sampler for this tracer provider. This must
    * not be a nullptr.
    */
-  explicit TracerProvider(std::unique_ptr<SpanProcessor> processor,
-                          opentelemetry::sdk::resource::Resource resource =
-                              opentelemetry::sdk::resource::Resource::Create({}),
-                          std::unique_ptr<Sampler> sampler =
-                              std::unique_ptr<AlwaysOnSampler>(new AlwaysOnSampler)) noexcept;
+  explicit TracerProvider(
+      std::unique_ptr<SpanProcessor> processor,
+      opentelemetry::sdk::resource::Resource resource =
+          opentelemetry::sdk::resource::Resource::Create({}),
+      std::unique_ptr<Sampler> sampler = std::unique_ptr<AlwaysOnSampler>(new AlwaysOnSampler),
+      std::unique_ptr<opentelemetry::sdk::trace::IdGenerator> id_generator =
+          std::unique_ptr<opentelemetry::sdk::trace::IdGenerator>(
```

Comment:
Is it better to make `IdGenerator` a global static? Is there any scenario to assign a `TraceProvider` a dedicated `IdGenerator`?

Reply:
As per the [specs](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#tracer-creation), `IdGenerator` should be managed by TracerProvider
``
Configuration (i.e., SpanProcessors, IdGenerator, SpanLimits and Sampler) MUST be managed solely by the TracerProvider and it MUST provide some way to configure all of them that are implemented in the SDK, at least when creating or initializing it.
``